test/jdk/com/sun/management/DiagnosticCommandMBean/DcmdMBeanTestCheckJni.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug     8258836
+ * @summary JNI local refs exceed capacity getDiagnosticCommandInfo
+ * @library /test/lib
+ * @build DcmdMBeanRunner
+ * @run main/othervm DcmdMBeanTestCheckJni
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class DcmdMBeanTestCheckJni {
+
+    public static void main(String[] args) throws Exception {
```

Comment:
I thought that handling the whole test in one file is the good style. See for example `
test/hotspot/jtreg/gc/stress/TestStressG1Uncommit.java`.

Reply:
OK. Updated the test.